### ISSUE TITLE
Numeric only tags create broken record

### ISSUE THREAD

**Issue by mediabeastnz** [silverstripe/silverstripe-tagfield#72]
If you currently enter a tag that is just numeric it doesn't save a tag in the has_one table but it does create a record in the many_many table that has a TagID of the title you entered e.g. 123. 

This means you'll have records in the many_many table that may not link to anything unless the numeric tag you entered somehow matches an existing record.

**Comment by tractorcow** [comment]
The problem is that the field uses is_numeric to distinguish between new tags and existing tags; A new schema will have to be developed. Maybe some kind of quoting mechanism so that all string values are posted back with quoted strings?

e.g.

```php
[
	'"one"', // new string
	'"1"', // literal 1
	'1', // existing key ID = 1
]
```

**Comment by mediabeastnz** [comment]
Where would you do this? Sorry i'm not that familiar with how it's all being saved.

Would you do something like you suggested inside the saveInto() function?

**Comment by tractorcow** [comment]
I'm honestly not sure; One would have to become familiar with select2.js to figure out a way of distinguishing between values that are typed in, and values that are being returned back un-touched.

**Comment by tractorcow** [comment]
@assertchris do you want to validate my idea is good? Or does it break something I haven't thought of?

**Comment by mediabeastnz** [comment]
Let me know what you think @assertchris and i'll try and submit a pull request. Need it urgently for a project :| 

Surprised no one has come across this before...

**Comment by tractorcow** [comment]
We just don't use numbers as tags. :)

**Comment by assertchris** [comment]
Sorry for the lag - I'm travelling to SunshinePHP. Probably won't get to this in days, but I trust @tractorcow to make a good decision here. He's my open source crush <3

**Comment by tractorcow** [comment]
Aww shucks. I'm going to let @mediabeastnz do the work for now though, since I unfortunately don't have time to work on this.

hope SunshinePHP is excellent.

**Comment by mediabeastnz** [comment]
No worries, I think I have it working. Just about to submit it now.

**Comment by mediabeastnz** [comment]
Just submitted a PR https://github.com/silverstripe-labs/silverstripe-tagfield/pull/71 

I haven't looked at StringTagField though so this might need to be checked as well? Although at first glance it looks very different and probably allows numeric items.

**Comment by robbieaverill** [comment]
Looks like fixed in #71, feel free to reopen if not
